src/core/src/function.cpp

```diff
@@ -913,3 +913,122 @@ ov::Output<ov::Node> ov::Function::add_output(const ov::Output<ov::Node>& port)
+        stream << "name='" << node.get_tensor().get_any_name() << "', ";
+    }
+    stream << "shape=" << node.get_partial_shape();
+    if (node.get_partial_shape().rank().is_static()) {
+        stream << ", layout=" << p->get_layout().to_string();
+        if (!ov::layout::has_batch(p->get_layout())) {
+            stream << ", no batch specified";
+        } else {
+            stream << ", batch="
+                   << node.get_partial_shape()[bs_util::get_batch(p->get_layout(), node.get_partial_shape())];
+        }
+        stream << " }" << std::endl;
+    }
+}
+}  // namespace bs_util
+
+ov::Dimension ov::util::get_batch_size(const std::shared_ptr<const ov::Function>& f) {
+    bool batch_initialized = false;
+    auto batch_size = ov::Dimension::dynamic();
+    std::vector<size_t> merged_indexes;
+    merged_indexes.reserve(f->inputs().size());
+    for (size_t i = 0; i < f->get_parameters().size(); ++i) {
+        const auto& param = f->get_parameters()[i];
+        const auto& layout = param->get_layout();
+        if (!ov::layout::has_batch(layout))
+            continue;
+        const auto& pshape = param->get_partial_shape();
+        if (pshape.rank().is_dynamic()) {
+            continue;  // Parameter with fully dynamic rank can't conflict
+        }
+        auto batch_idx = bs_util::get_batch(layout, pshape);
+        if (!Dimension::merge(batch_size, batch_size, pshape[batch_idx])) {
+            merged_indexes.push_back(i);
+            // Not all dimensions can be merged
+            std::stringstream stream;
+            stream << "Get original batch size fails due to conflicting batch values for inputs:" << std::endl;
+            for (size_t j = 0; j < merged_indexes.size(); ++j) {
+                bs_util::dump_parameter(stream, f, merged_indexes[j]);
+            }
+            stream << "---" << std::endl;
+            stream << "Please ensure that N(Batch) dimension is set correctly for listed parameters";
+            OPENVINO_ASSERT(false, stream.str());
+        } else {
+            merged_indexes.push_back(i);
+        }
+        batch_initialized = true;
+    }
+    if (!batch_initialized) {
+        // Create graceful message to set layout for some parameters
+        std::stringstream stream;
+        stream << "Get original batch size fails due to batch is not set in any layout for any input. ";
+        stream << "Available inputs:" << std::endl;
+        for (size_t i = 0; i < f->get_parameters().size(); ++i) {
+            bs_util::dump_parameter(stream, f, i);
+        }
+        stream << "---" << std::endl;
+        stream << "Please use 'set_layout' API to set layout with batch dimension, e.g. "
+                  "`Function->get_parameters()[index]->set_layout(\"NCHW\");`";
+
+        OPENVINO_ASSERT(false, stream.str());
+    }
+    return batch_size;
+}
+
+void ov::util::set_batch_size(const std::shared_ptr<ov::Function>& f, ov::Dimension batch_size) {
```

Comment:
BTW, can we reuse it in existing samples like https://github.com/openvinotoolkit/openvino/blob/master/samples/cpp/ngraph_function_creation_sample/main.cpp#L325

Maybe in a separate PR.

Reply:
added to this pr